src/diff/index.js

```diff
@@ -171,7 +167,7 @@ export function commitRoot(mounts, root) {
 			c.componentDidMount();
 		}
 		catch (e) {
-			catchErrorInComponent(e, c._ancestorComponent);
+			catchErrorInComponent(e, c._vnode);
```

Comment:
isn't this missing `._parent`?

Seems like this could make the component be able to catch it's own error which I'm not sure is possible in react

Reply:
That's true! I'll add a test and fix this

Reply:
Interesting... Does appear that in React, a component can catch it's own ref though: https://codesandbox.io/s/react-throwing-refs-lk958 (that actually matches the behavior in this branch already lol)

Reply:
Yes a component can catch errors in its own render/lifecycles, I think we're aligning more with React like this.

Reply:
Hmmm this codesandbox seems to show that a component can't catch errors in it's own lifecyles. Does this look right? https://codesandbox.io/s/react-throwing-lifecycles-kjwky

Reply:
TIL about react catching `ref` errors in the same component. Apart from that I'm not aware of components being able to catch errors inside their own lifecycle functions like the codesandbox demonstrates :+1: